apps/full-site-editing/full-site-editing-plugin/global-styles/includes/class-global-styles-json-endpoint.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * JSON REST API endpoint for Global Styles plugin.
+ */
+
+namespace A8C\Global_Styles;
+
+/**
+ * REST API endpoint for Global Styles plugin.
+ */
+class JSON_Endpoint extends \WP_REST_Controller {
+
+	/**
+	 * Namespace for the REST endpoint.
+	 *
+	 * @var string
+	 */
+	private $rest_namespace;
+
+	/**
+	 * Route name for the REST endpoint.
+	 *
+	 * @var string
+	 */
+	private $rest_route;
+
+	/**
+	 * Object holding the data description to work with.
+	 *
+	 * @var A8C\Global_Styles\Data_Set
+	 */
+	private $data_set;
+
+	/**
+	 * Constructor
+	 *
+	 * @param string $rest_namespace Namespace for the REST endpoint.
+	 * @param string $rest_route Route name.
+	 * @param array  $data_set Description of the data to work with.
+	 */
+	public function __construct( $rest_namespace, $rest_route, $data_set ) {
+		$this->rest_namespace = $rest_namespace;
+		$this->rest_route     = $rest_route;
+		$this->data_set       = $data_set;
+	}
+
+	/**
+	 * Callback to determine whether the request can proceed.
+	 *
+	 * @return boolean
+	 */
+	public function permission_callback() {
+		return is_user_logged_in() && apply_filters( 'global_styles_permission_check_additional', true );
```

Comment:
I'm a bit suspicious allowing the only check to be `is_user_logged_in()` instead of having capability checks like `edit_posts`.

Reply:
The permission mechanism was something I had in my queue to address. It now checks for `customize` capability in addition to theme support, etc - https://github.com/Automattic/wp-calypso/pull/36288/commits/9ab0244ea6d070a45331feb5d9757d8675317219